src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -48,27 +48,23 @@ public class UserManager<TUser> : IDisposable where TUser : class
         /// <param name="contextAccessor">The accessor used to access the <see cref="HttpContext"/>.</param>
         public UserManager(IUserStore<TUser> store,
             IOptions<IdentityOptions> optionsAccessor,
-            IPasswordHasher<TUser> passwordHasher,
-            IEnumerable<IUserValidator<TUser>> userValidators,
-            IEnumerable<IPasswordValidator<TUser>> passwordValidators,
-            ILookupNormalizer keyNormalizer,
-            IdentityErrorDescriber errors,
             IServiceProvider services,
-            ILogger<UserManager<TUser>> logger,
-            IHttpContextAccessor contextAccessor)
+            ILogger<UserManager<TUser>> logger)
         {
             if (store == null)
             {
                 throw new ArgumentNullException(nameof(store));
             }
             Store = store;
-            Options = optionsAccessor?.Value ?? new IdentityOptions();
-            _context = contextAccessor?.HttpContext;
-            PasswordHasher = passwordHasher;
-            KeyNormalizer = keyNormalizer;
-            ErrorDescriber = errors;
             Logger = logger;
+            Options = optionsAccessor?.Value ?? new IdentityOptions();
+
+            _context = services?.GetService<IHttpContextAccessor>()?.HttpContext;
+            PasswordHasher = services?.GetService<IPasswordHasher<TUser>>() ?? new PasswordHasher<TUser>();
+            KeyNormalizer = services?.GetService<ILookupNormalizer>() ?? new UpperInvariantLookupNormalizer(); ;
```

Comment:
If we do care about new() support, This is what I'd propose for the ctor:

```UserManager(IUserStore<TUserStore> store, IServiceProvider services)```

Where store is required and must not be null, and we'd look up everything else from the services, and use defaults if needed.  AddIdentity would actually no longer need to register any of these worker services.  It would only register the top level mangers (User/Role/SignIn)